File: mobile/src/main/java/net/mm2d/dmsexplorer/util/FeatureUtils.java
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.dmsexplorer.util;

import android.content.Context;
import android.content.pm.PackageManager;
import android.support.annotation.NonNull;

/**
 * @author <a href="mailto:ryo@mm2d.net">大前良介 (OHMAE Ryosuke)</a>
 */
public class FeatureUtils {
    public static boolean hasTouchScreen(@NonNull final Context context) {
        final PackageManager pm = context.getPackageManager();
        return pm.hasSystemFeature(PackageManager.FEATURE_TOUCHSCREEN);
    }
}

File: mobile/src/main/java/net/mm2d/dmsexplorer/util/FeatureUtils.kt
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.dmsexplorer.util

import android.content.Context
import android.content.pm.PackageManager

/**
 * @author [大前良介 (OHMAE Ryosuke)](mailto:ryo@mm2d.net)
 */
object FeatureUtils {
    @JvmStatic
    fun hasTouchScreen(context: Context): Boolean {
        return context.packageManager
                .hasSystemFeature(PackageManager.FEATURE_TOUCHSCREEN)
    }
}
